testsuite/features/step_definitions/command_steps.rb

```diff
@@ -179,6 +179,17 @@
   $command_output, _code = get_target('server').run(command)
 end
 
+When(/^I use spacewalk-common-channel to add all "([^"]*)" channels with arch "([^"]*)"$/) do |channel, architecture|
+  channels_to_synchronize = CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, "#{channel}-#{architecture}")
+  raise ScriptError, "Synchronization error, version type #{channel}-#{architecture} in #{product} product not found" if channels_to_synchronize.nil?
+
+  channels_to_synchronize.each do |os_product_version_channel|
+    log "Adding channel: #{os_product_version_channel}"
+    command = "spacewalk-common-channels -u admin -p admin -a #{architecture} #{os_product_version_channel.gsub("-#{architecture}", '')}"
```

Comment:
`delete_suffix()` might be a bit more legible than `gsub()`

Reply:
Taking the example:
```
        opensuse_leap15_5-x86_64
        opensuse_leap15_5-x86_64-non-oss
        opensuse_leap15_5-x86_64-non-oss-updates
        opensuse_leap15_5-x86_64-updates
```
the `x86_64` is not always in the end of given channel, so `delete_suffix` method is not applicable here.

Reply:
ah! ok